client/lib/Hero.ts

```diff
@@ -481,12 +475,17 @@ export default class Hero extends AwaitedEventTarget<{
     return this.activeTab.waitForMillis(millis);
   }
 
-  public async waitForPageState<T extends IPageStateDefinitions>(
-    states?: T,
+  public async waitForState(
+    state: IDomState | DomState,
     options?: Pick<IWaitForOptions, 'timeoutMs'>,
-  ): Promise<keyof T> {
-    return await this.activeTab.waitForPageState(states, options);
+  ): Promise<boolean> {
+    return await this.activeTab.waitForState(state, options);
   }
+
+  public async ensureState(state: IDomState | DomState): Promise<boolean> {
+    return await this.activeTab.ensureState(state);
+  }
```

Comment:
We had discussed changing this to `checkState`. I'm okay with either one, just curious your thoughts.

Reply:
I have it changed in the new branch